Testing hr-automation-api

zhenekq opened the pull request.

zhenekq committed:
started files, don't know is working correctly

zhenekq committed:
try hard

misyukevich-itransition committed:
test commit

zhenekq committed:
test commit123

zhenekq committed:
Tested candidate controller test

zhenekq committed:
Tested communication history controller and some fixes

zhenekq committed:
Attribute types tests completed and some fixes

zhenekq committed:
Candidate attributes test created and some fixes

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/AttributeTypesController.java:
```diff
@@ -22,7 +22,7 @@
 
 @RestController
-@RequestMapping("/attributestypes")
+    @RequestMapping("/attributetypes")
```
What is the meanning of that space before annotation?

zhenekq replied:
fixed, missclick

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/dto/CandidateDto.java:
```diff
@@ -107,6 +113,10 @@ public int hashCode() {
     @Override
     public String toString() {
-        return ToStringBuilder.reflectionToString(this);
+        return "CandidateDto{" +
```
Example
return new ToStringBuilder(this).
       append("name", name).
       append("age", age).
       append("smoker", smoker).
       toString();

zhenekq replied:
fixed missclicked

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/dto/CommunicationHistoryDto.java:
```diff
@@ -77,6 +77,12 @@ public int hashCode() {
     public String toString() {
-        return ToStringBuilder.reflectionToString(this);
+        return "CommunicationHistoryDto{" +
+                "id=" + id +
```
See example above

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/Candidate.java:
```diff
@@ -37,31 +36,31 @@
      * @see CommunicationHistory
      */
-    @OneToMany(mappedBy = "candidate")
+    @OneToMany(mappedBy = "candidate", fetch = FetchType.LAZY)
```
look at source code. fetch is lazy by default

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/Candidate.java:
```diff
@@ -37,31 +36,31 @@
      * @see CandidateAttributes
      */
-    @OneToMany(mappedBy = "candidate")
+    @OneToMany(mappedBy = "candidate", fetch = FetchType.LAZY)
```
look at source code. fetch is lazy by default

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/Keyword.java:
```diff
@@ -65,6 +65,8 @@ public int hashCode() {
     @Override
     public String toString() {
-        return ToStringBuilder.reflectionToString(this);
+        return "Keyword{" +
```
See comment above

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/util/validator/impl/AttributeTypeValidator.java:
```diff
@@ -0,0 +1,16 @@
+public class AttributeTypeValidator implements EntityValidator<AttributeTypes> {
+    @Override
+    public boolean isValidParams(AttributeTypes attributeTypes) {
+        return  attributeTypes.getBasicType() != null &&
```
Are theses fields can be empty for validation?

Maybe look at Bean Validation

AndrewRest commented on src/test/java/by/mifort/automation/hr/dev/controller/CandidateAttributeControllerTest.java:
```diff
@@ -0,0 +1,145 @@
+    private final AttributeTypesController typesController;
+    private final EntityConverter<CandidateAttributes, CandidateAttributesDto> converter;
+
+    private final H2Database h2Database = H2Database.getInstance();
```
If you are doing the same things for several tests - make parent test

AndrewRest commented on src/test/java/by/mifort/automation/hr/dev/controller/CandidateControllerTest.java:
```diff
@@ -0,0 +1,142 @@
+    @Test
+    @DisplayName("Check are connected keyboards will be ignored when connected away, with new")
+    void addKeywordsToCandidateWithKeywords(){
+        List<String> addedKeywords = new ArrayList<>(List.of("java", "react"));
```
List.of already returns List why do you need ArrayList envelop?

zhenekq committed:
big fixes

zhenekq committed:
fixes

zhenekq committed:
created some basic logic for finding duplicates with using just one parameter and some javadoc for it

zhenekq committed:
created attributetypes database structure with attributes

zhenekq committed:
fixed adding candidate attributes with many parameters

zhenekq committed:
fixed return list of duplicates

zhenekq committed:
fixed strategy logic

zhenekq committed:
Create functionality to find by:
    1) Firstname, Lastname, City, Date of birth
    2) Firstname, Lastname, City, Company

zhenekq committed:
Added functionality + created database for testing + fixed test for attributes

zhenekq committed:
Fixed tests for candidateAttributes

zhenekq committed:
new logic of finding by one attribute

zhenekq committed:
FINAL FIX TESTS

zhenekq committed:
Fixed isArchived column from AttributeTypes to CandidateAttributes

zhenekq committed:
Little validation for create user

zhenekq committed:
dont care

zhenekq committed:
Merge pull request #7 from zhenekq/find-by-many-attributes

multi find attributes

zhenekq committed:
Merge pull request #6 from zhenekq/merge-find-duplicates-dev

Merge / find duplicates
